src/Libraries/RevitNodes/Elements/DimensionType.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Linq;
+using Autodesk.DesignScript.Runtime;
+using DynamoServices;
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+
+namespace Revit.Elements
+{
+    /// <summary>
+    /// Dimension Type element.
+    /// </summary>
+    [RegisterForTrace]
+    public class DimensionType : Element
+    {
+        #region Internal Properties
+
+        /// <summary>
+        /// Internal reference to the Revit Element
+        /// </summary>
+        internal Autodesk.Revit.DB.DimensionType InternalRevitElement
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Reference to the Element
+        /// </summary>
+        [SupressImportIntoVM]
+        public override Autodesk.Revit.DB.Element InternalElement
+        {
+            get { return InternalRevitElement; }
+        }
+
+        #endregion
+
+        #region Private mutators
+
+        /// <summary>
+        /// Set the internal Element, ElementId, and UniqueId
+        /// </summary>
+        /// <param name="element"></param>
+        private void InternalSetElement(Autodesk.Revit.DB.DimensionType element)
+        {
+            InternalRevitElement = element;
+            InternalElementId = element.Id;
+            InternalUniqueId = element.UniqueId;
+        }
+
+        #endregion
+
+        #region Private constructors
+
+        /// <summary>
+        /// DimensionType from existing
+        /// </summary>
+        /// <param name="element"></param>
+        private DimensionType(Autodesk.Revit.DB.DimensionType element)
+        {
+            SafeInit(() => InitElement(element));
+        }
+
+        /// <summary>
+        /// DimensionType by duplicating existing.
+        /// </summary>
+        /// <param name="dimType"></param>
+        /// <param name="name"></param>
+        private DimensionType(Autodesk.Revit.DB.DimensionType dimType, string name)
+        {
+            SafeInit(() => Init(dimType, name));
+        }
+
+        #endregion
+
+        #region Helpers for private constructors
+
+        private void InitElement(Autodesk.Revit.DB.DimensionType element)
+        {
+            InternalSetElement(element);
+        }
+
+        private void Init(Autodesk.Revit.DB.DimensionType dimType, string name)
+        {
+            var document = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(document);
+
+            // get element from trace
+            var element = ElementBinder.GetElementFromTrace<Autodesk.Revit.DB.DimensionType>(document);
+
+            if (element == null)
+            {
+                element = dimType.Duplicate(name) as Autodesk.Revit.DB.DimensionType;
```

Comment:
@ksobon Does this mean each time user run the graph we unconditionally create a new DimensionType with the new name? Does not seem to scan through the existing DimensionTypes. I imagine it could cause trouble in certain cases. Please correct me if I was wrong.